src/execution_plan/optimizations/reduce_count.c

```diff
@@ -144,6 +144,36 @@ static int _identifyEdgeCountPattern(OpBase *root, OpResult **opResult, OpAggreg
 
     return 1;
 }
+
+void _accumulateEdges(void *_z, const void* _x, const void* _y) {
+    uint64_t* sum = (uint64_t*) _z;
+    uint64_t* currentTotal = (uint64_t*) _x;
+    const EdgeID *e = (const EdgeID*)_y;
+
+    EdgeID *ids;
+    // Single edge ID
+    if(SINGLE_EDGE(*e)) {
+        *sum = *currentTotal + 1;
+    } else {
+        // Multiple edges
+        ids = (EdgeID*)(*e);
+        *sum = *currentTotal + array_len(ids);
+    }
+}
+
+uint64_t _countRelationshipEdges(GrB_Matrix relationshipMatrix){
+    GrB_BinaryOp op;
+    GrB_Info info = GrB_BinaryOp_new(&op, _accumulateEdges, GrB_UINT64, GrB_UINT64, GrB_UINT64);
+
+    GrB_Monoid monoid;
```

Comment:
Rename to a more meaningful name than monoid.

Reply:
done